metaflow/plugins/metadata/local.py

```diff
@@ -112,6 +115,35 @@ def register_metadata(self, run_id, step_name, task_id, metadata):
         }
         self._save_meta(meta_dir, metadict)
 
+    @classmethod
+    def _mutate_user_tags_for_run(
```

Comment:
This operation is non atomic. We should guarantee sequential ordering of tagging operations IMHO

Reply:
Could you elaborate?  It looks atomic to me (We write to temporary file and then atomic rename it into `_self.json`).  Last write wins, all or nothing.

We will potentially drop concurrent updates.  E.g. two processes: one adds A, one adds B.... we may only end up seeing one of them in final result.  This is unavoidable without a serialization mechanism (e.g. locks).   IMO it is risky to do this, doesn't seem worth it here.  The Metadata service mutation operation will be atomic AND serializable (coming up soon).

Reply:
Discussed in this [slack thread](https://outerboundsco.slack.com/archives/C023Y9JH36W/p1653416648537399?thread_ts=1653349445.888399&cid=C023Y9JH36W).

As a way to provide a more predictable UX (no dropped changes) whilst avoiding locks, I have added a simple retry with backoff until the callers sees what it expects post operation.